sdk/communication/azure-communication-callautomation/azure/communication/callautomation/aio/_content_downloader_async.py

```diff
@@ -0,0 +1,169 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from typing import Any
+
+from  urllib.parse import ParseResult, urlparse
+from azure.core.exceptions import (
+    ClientAuthenticationError,
+    HttpResponseError,
+    ResourceExistsError,
+    ResourceNotFoundError,
+    ResourceNotModifiedError,
+    map_error,
+)
+from azure.core.pipeline import PipelineResponse
+from azure.core.pipeline.transport import HttpResponse
+from azure.core.rest import HttpRequest
+from azure.core.utils import case_insensitive_dict
+
+from .._generated import models as _models
+from .._generated._serialization import Serializer
+from .._generated.aio.operations import CallRecordingOperations
+
+_SERIALIZER = Serializer()
+_SERIALIZER.client_side_validation = False
+
+
+class ContentDownloader(object):
+    def __init__(
+            self,
+            call_recording_client:CallRecordingOperations
+        ) -> None:
+
+        self._call_recording_client = call_recording_client
+
+    async def download_streaming(  # pylint: disable=inconsistent-return-statements
+        self, source_location: str, offset: int, length: int, **kwargs: Any
+    ) -> HttpResponse:
+        """Download a stream of the call recording.
+
+        :param source_location: The source location. Required.
+        :type source_location: str
+        :param offset: Offset byte. Not required.
+        :type offset: int
+        :param length: how many bytes. Not required.
+        :type length: int
+        :return: HttpResponse (octet-stream)
+        :rtype: HttpResponse (octet-stream)
+        """
+
+        if length is not None and offset is None:
+            raise ValueError("Offset value must not be None if length is set.")
+        if length is not None:
+            length = offset + length - 1  # Service actually uses an end-range inclusive index
```

Comment:
According to the guideline, we should skip the parameter validation, however it may affect this logic, please consider what would be the best approach. Maybe do a validation but don't throw? And then use 0 to represent None.

Reply:
Check the sync method as well.

Reply:
I think the validation is effective only for server calls - this is local method, which we must do validation here (there is no "server" to do validation)